201-encrypt-running-windows-vm/azuredeploy.json

```diff
@@ -26,120 +14,137 @@
     },
-    "volumeType": {
-      "type": "string",
-      "defaultValue": "All",
-      "metadata": {
-        "description": "Type of the volume OS or Data to perform encryption operation"
-      }
-    },
-    "sequenceVersion": {
-      "type": "string",
-      "defaultValue": "1.0",
-      "metadata": {
-        "description": "Pass in an unique value like a GUID everytime the operation needs to be force run"
-      }
-    },
-    "location": {
-      "type": "string",
-      "defaultValue": "[resourceGroup().location]",
-      "metadata": {
-        "description": "Location for all resources."
+      "enabledForDeployment": {
+        "type": "bool",
+        "defaultValue": true,
+        "allowedValues": [
+          true,
+          false
+        ],
+        "metadata": {
+          "description": "Specifies whether Azure Virtual Machines are permitted to retrieve certificates stored as secrets from the key vault."
+        }
+      },
+      "enabledForDiskEncryption": {
+        "type": "bool",
+        "defaultValue": true,
+        "allowedValues": [
+          true,
+          false
+        ],
+        "metadata": {
+          "description": "Specifies whether Azure Disk Encryption is permitted to retrieve secrets from the vault and unwrap keys."
+        }
+      },
+      "enabledForTemplateDeployment": {
+        "type": "bool",
+        "defaultValue": true,
+        "allowedValues": [
+          true,
+          false
+        ],
+        "metadata": {
+          "description": "Specifies whether Azure Resource Manager is permitted to retrieve secrets from the key vault."
+        }
+      },
+      "keysPermissions": {
+        "type": "array",
+        "defaultValue": [
+          "list",
+          "get",
+          "decrypt",
+          "encrypt",
+          "unwrapkey",
+          "wrapkey"
+        ],
+        "metadata": {
+          "description": "Specifies the permissions to keys in the vault. Valid values are: all, encrypt, decrypt, wrapKey, unwrapKey, sign, verify, get, list, create, update, import, delete, backup, restore, recover, and purge."
+        }
+      },
+      "secretsPermissions": {
+        "type": "array",
+        "defaultValue": [
+          "list",
+          "get"
+        ],
+        "metadata": {
+          "description": "Specifies the permissions to secrets in the vault. Valid values are: all, get, list, set, delete, backup, restore, recover, and purge."
+        }
+      },
+      "skuName": {
+        "type": "string",
+        "defaultValue": "Standard",
+        "allowedValues": [
+          "Standard",
+          "Premium"
+        ],
+        "metadata": {
+          "description": "Specifies whether the key vault is a standard vault or a premium vault."
+        }
       }
-    }
   },
   "variables": {
-    "extensionName": "AzureDiskEncryption",
-    "extensionVersion": "1.1",
+    "keyVaultURL": "[concat('https://', parameters('keyVaultName'), parameters('keyVaultDomain'))]",
```

Comment:
"keyVaultURL": "[concat('https://', parameters('keyVaultName'), '.', environment().suffixes.keyVaultDns)]",

Reply:
thanks, didnt know theres one :)

Reply:
doesnt seem to be working with the dot,

"[2.2.0.39] Failed to configure bitlocker as expected. Exception: Invalid URI https://cia3a1b3..vault.azure.net"